src/org/rascalmpl/compiler/lang/rascalcore/package/Packager.rsc

```diff
@@ -70,10 +70,14 @@ value rewriteTypeModel(value model, map[loc,str] paths, loc sourceLookup)
           // any location in the wild:
           case loc l => inheritPosition(sourceLookup + paths[l.top], l)
               when l.top in paths
-            
+
           // \loc annotations on Trees are not visited by `visit` automatically
           case Tree t => t[@\loc = inheritPosition(sourceLookup + paths[Top], t@\loc)]
               when t@\loc?, loc Top := t@\loc.top, Top in paths
+
+          // remove infos and warnings
+          case {Message _, *_} => {}
+          case [Message _, *_] => []
```

Comment:
what if there's a list or set that contains multiple values, both messages and other things? now we throw them away? I would think that `list[Message] => []`  is more accurate of a pattern?

Reply:
This matches at least a singleton and therefore skips all the empty lists, which may result in much fast `visit` behavior since it will not rebuild the tree if nothing changed.